arangod/Indexes/SortedIndexAttributeMatcher.cpp

```diff
@@ -278,13 +278,6 @@ Index::FilterCosts SortedIndexAttributeMatcher::supportsFilterCondition(
     std::vector<std::shared_ptr<arangodb::Index>> const& allIndexes,
     arangodb::Index const* idx, arangodb::aql::AstNode const* node,
     arangodb::aql::Variable const* reference, size_t itemsInIndex) {
-  // mmfiles failure point compat
-  if (idx->type() == Index::TRI_IDX_TYPE_HASH_INDEX) {
-    TRI_IF_FAILURE("SimpleAttributeMatcher::accessFitsIndex") {
-      THROW_ARANGO_EXCEPTION(TRI_ERROR_DEBUG);
-    }
-  }
-
```

Comment:
Removed from here because this method is index type-agnostic and should not carry out code for specific index types. Moved it to `RocksDBVPackIndex` instead, which is the base class for the `RocksDBHashIndex` class (`RocksDBHashIndex` inherits almost all behavior from `RocksDBHashIndex`).